### DISCUSSION TITLE
Performance cost of attribute splatting

### DISCUSSION THREAD

**Discussion by stefanloerwald** [dotnet/aspnetcore#27059] [Q&A] [answered]
Hi,

a question came to my mind reading [webassembly-performance-best-practices](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/webassembly-performance-best-practices.md#avoid-attribute-splatting-with-captureunmatchedvalues):

Is the performance cost for `CaptureUnmatchedValues` only there if there are unmatched values, or is it there as soon as there is such a parameter?

Thanks
Stefan

**Comment by stefanloerwald** [comment, marked as answer]
@SteveSandersonMS maybe you could say something about this? Thanks!

**Reply by SteveSandersonMS** [reply]
The expense is mostly on the outbound side, i.e., when you write `@attributes=SomeDictionary` in your component output to pass through the unmatched parameters into output, because to render all subsequent attributes on the same element, it has to check whether or not it matches an existing entry in the dictionary. This is true even if `SomeDictionary` happens to be empty, because the code doesn't know it will *always* be empty.

On the inbound side, the expense is only for unmatched parameters, because those are the only ones added to the dictionary.
